Assets/General/Scripts/ScreenText/ScriptableScreenTextUI.cs

```diff
@@ -0,0 +1,164 @@
+/*****************************************************************************
+// File Name :         ScriptableScreenTextUI.cs
+// Author :            Nick Rice
+//                     
+// Creation Date :     10/21/24
+//
+// Brief Description : This script handles data containers for readable UI elements in the game
+*****************************************************************************/
+
+using System;
+using Unity.VisualScripting;
+using UnityEngine;
+
+/// <summary>
+/// The scriptable object for tutorial text and timing
+/// </summary>
+[CreateAssetMenu(menuName = "UI Data Object/TutorialData")]
+public class ScriptableTutorialUI : ScriptableObject
+{
+    [Tooltip("Tutorial text and time variables")]
+    [SerializeField]
+    private TextAndTimerData _tutorialData;
+    
+    #region Getting Total Time
+    [Tooltip("How much time the tutorial takes to display")]
+    public uint TotalTime { get; private set; }
+
+    /// <summary>
+    /// Checks to see if it's null, then performs total time operation
+    /// </summary>
+    private void Awake()
+    {
+        _tutorialData.GetText.NullIfEmpty();
+        if (IsActive())
+        {
+            TotalTime  = _tutorialData.GetTimeBeforeText + _tutorialData.GetTimeToDisplay;
+        }
+    }
+    #endregion
+    
+    /// <summary>
+    /// Checks to see if the dialogue data is not null or unused
+    /// </summary>
+    public bool IsActive()
+    {
+        bool active = false;
+
+        active = !_tutorialData.GetText.IsUnityNull() && _tutorialData.GetText != "";
+        
+        return active;
+    }
+    
+    /// <summary>
+    /// Returns a reference to tutorial data
+    /// </summary>
+    public TextAndTimerData GetTextAndTimer()
+    {
+        return _tutorialData;
+    }
+}
+
+/// <summary>
+/// The scriptable object for dialogue text and timing
+/// </summary>
+[CreateAssetMenu(menuName = "UI Data Object/DialogueData")]
+public class ScriptableDialogueUI : ScriptableObject
+{
+    [Tooltip("Dialogue text and time variables")]
+    [SerializeField]
+    private TextAndTimerData[] _dialogueData;
+
+    #region Getting Total Time
+    [Tooltip("How much time a chain of dialogue takes to display")]
+    public uint TotalTime { get; private set; }
+
+    /// <summary>
+    /// Checks to see if it's null, then performs total time operation
+    /// </summary>
+    private void Awake()
+    {
+        _dialogueData[0].GetText.NullIfEmpty();
+        if (IsActive())
+        {
+            for (int i = 0; i < _dialogueData.Length; i++)
```

Comment:
For cleanliness sake could this be done in it's own function and just have awake call that. Makes sure that Awake is easier to read.